src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -0,0 +1,143 @@
+using NUnit.Framework;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Xamarin.ProjectTools;
+using XABuildPaths = Xamarin.Android.Build.Paths;
+
+namespace Xamarin.Android.Build.Tests
+{
+	public class DeviceTest: BaseTest
+	{
+		protected static void RunAdbInput (string command, params object [] args)
+		{
+			RunAdbCommand ($"shell {command} {string.Join (" ", args)}");
+		}
+
+		protected static string ClearAdbLogcat ()
+		{
+			return RunAdbCommand ("logcat -c");
+		}
+
+		protected static string GetAdbLogcat (int timeout = 1)
+		{
+			return RunAdbCommand ("logcat", timeout: timeout);
+		}
```

Comment:
I think we should remove this method, mainly because we don't want enormous logcat output as a `string`.